src/universal_transfer_operator/integrations/fivetran/connector/aws/s3.py

```diff
@@ -0,0 +1,292 @@
+            trust_fingerprints=self.trust_fingerprints,
+            run_setup_tests=self.run_setup_tests,
+            connect_card_config=self.connect_card_config,
+        )
+
+    def to_dict(self) -> dict:
+        """
+        Convert options class to dict
+        """
+        return vars(self)
+
+    def _hook(self, conn_id: str, verify: Any, transfer_config_args: Any, s3_extra_args: Any) -> S3Hook:
+        """Return an instance of the database-specific Airflow hook."""
+        return S3Hook(
+            aws_conn_id=conn_id,
+            verify=verify,
+            transfer_config_args=transfer_config_args,
+            extra_args=s3_extra_args,
+        )
+
+    def map_airflow_connection_to_fivetran(
+        self,
+        source_dataset: Dataset,
+        destination_dataset: Dataset,
+        group_id: str,
+    ) -> dict[str, str]:
+        """
+        Maps the airflow connection details to fivetran config for destination
+
+        :param source_dataset: Source dataset
+        :param destination_dataset: Destination dataset
+        :param group_id: Group id in fivetran system
+        """
+        conn = BaseHook().get_connection(source_dataset.conn_id)  # type: ignore
+
+        access_key = conn.login
+        secret_key = conn.password
+
+        if not access_key:
+            raise ConnectionError(f"AWS_ACCESS_KEY_ID is not configured for {source_dataset.conn_id}")  # type: ignore
+
+        if not secret_key:
+            raise ConnectionError(f"AWS_SECRET_ACCESS_KEY is not configured {source_dataset.conn_id}")  # type: ignore
+
+        s3_hook = self._hook(
+            conn_id=source_dataset.conn_id,  # type: ignore
+            verify=self._verify(source_dataset),
+            transfer_config_args=self._transfer_config_args(source_dataset),
+            s3_extra_args=self._s3_extra_args(source_dataset),
+        )
+
+        bucket_name = self._bucket_name(s3_hook=s3_hook, dataset=source_dataset)
+        # Create IAM policy. Read more at: Read more at: https://fivetran.com/docs/files/amazon-s3/setup-guide
+        iam_policy = self._create_aws_policy(
+            bucket_name=bucket_name, access_key=access_key, secret_key=secret_key
+        )
+
+        # Create IAM Role. Read more at: Read more at: https://fivetran.com/docs/files/amazon-s3/setup-guide
+        iam_role = self._create_role(
+            role_name=self._generate_unique_fivetran_name(),
+            fivetran_aws_vpc_account_id=str(FIVETRAN_AWS_VPC_ACCOUNT_ID),
+            group_id=group_id,
+            access_key=access_key,
+            secret_key=secret_key,
+        )
+
+        # Attach IAM policy to role. Read more at: Read more at: https://fivetran.com/docs/files/amazon-s3/setup-guide
+        self._attach_to_role(
+            role_name=iam_role["Role"]["RoleName"],
+            policy_arn=iam_policy["Policy"]["Arn"],
+            access_key=access_key,
+            secret_key=secret_key,
+        )
+
+        connection_details = {
```

Comment:
Did we abandon the idea of using the patterns?

Reply:
Yes, @utkarsharma2 that was not required in this scenario at all as configuration was just required to be passed for API